src/searchModal/advancedSearch.js

```diff
@@ -131,30 +134,41 @@ export default function advancedSearchFactory(config) {
      * Inits select2 on criteria select and its UX logic
      */
     function initAddCriteriaSelector() {
-        $criteriaSelect.select2({
-            containerCssClass: 'criteria-select2',
-            dropdownCssClass: 'criteria-dropdown-select2',
-            sortResults: results => _.sortBy(results, ['text'])
-        });
-
-        // open dropdown when user clicks on add criteria input
-        $addCriteriaInput.on('click', () => {
-            $criteriaSelect.select2('open');
-            // if dropdown is opened above addCriteria input, top property is slightly decreased to avoid overlapping with addCriteria icon
-            if ($('.criteria-dropdown-select2').hasClass('select2-drop-above')) {
-                $('.criteria-dropdown-select2').css(
-                    'top',
-                    $('.criteria-dropdown-select2').css('top').split('px')[0] - 10 + 'px'
-                );
+        const route = urlUtil.route('status', 'AdvancedSearch', 'tao');
+        request(route).then(function (response) {
+            if (!response.enabled) {
+                $addCriteria.addClass('disabled');
```

Comment:
Considering that you have already added `disabled` class to `.add-criteria-container` on  `initUiSelectors`, which is executed right before `initAddCriteriaSelector`, and that this happens only once (on `render` event), I don't see when would be necessary to add `disabled` class agian.